### PR TITLE
enable control over action modal by option to pass defaults array

### PR DESCRIPTION
**Pull request by georgehristov**

### CODE REVIEW

File: src/TableColumn/ActionButtons.php [hunk 1]
```diff
@@ -100,7 +100,7 @@ public function addButton($button, $action = null, $confirm = false, $isDisabled
      * load contents through $callback. Will pass a virtual page.
      *
      * @param \atk4\ui\View|string $button
-     * @param string               $title
+     * @param string|array         $title - model title or model seed array
```
Comment (mvorisek) [inline src/TableColumn/ActionButtons.php]: There should be no `-`.

"model", what class is seeded by default?

File: src/TableColumn/ActionButtons.php [hunk 2]
```diff
@@ -111,7 +111,7 @@ public function addModal($button, $title, $callback, $owner = null, $args = [])
     {
         $owner = $owner ?: $this->owner->owner;
 
-        $modal = \atk4\ui\Modal::addTo($owner, [compact('title'), 'appStickyCb' => true]);
+        $modal = \atk4\ui\Modal::addTo($owner, [is_array($title) ? $title : compact('title'), 'appStickyCb' => true]);
```
Comment (mvorisek) [inline src/TableColumn/ActionButtons.php]: Replace all `compact('xx', ...)` usages to `['xx' => $xx, ...]`, compact is not refactorable.

**Review by mvorisek — CHANGES_REQUESTED** [review]

File: src/TableColumn/ActionButtons.php (hunk 2 above)
Comment (georgehristov) [reply src/TableColumn/ActionButtons.php]: Corrected

File: src/TableColumn/ActionButtons.php (hunk 1 above)
Comment (georgehristov) [reply src/TableColumn/ActionButtons.php]: Corrected

**Review by mvorisek — APPROVED** [review]

File: src/TableColumn/ActionButtons.php [hunk 3]
```diff
@@ -100,18 +100,20 @@ public function addButton($button, $action = null, $confirm = false, $isDisabled
...
      *
      * @return \atk4\ui\View
      */
-    public function addModal($button, $title, $callback, $owner = null, $args = [])
+    public function addModal($button, $defaults, $callback, $owner = null, $args = [])
     {
         $owner = $owner ?: $this->owner->owner;
 
-        $modal = \atk4\ui\Modal::addTo($owner, [compact('title'), 'appStickyCb' => true]);
+        $defaults = [is_array($defaults) ? $defaults : ['title' => $defaults], 'appStickyCb' => true];
```
Comment (mvorisek) [inline src/TableColumn/ActionButtons.php]: Array merge is probably needed, so this should look like:
```
if (is_string($defaults)) {
    $defaults = ['title' => $defaults];
}

$defaults['appStickyCb'] = true;
```

**Review by mvorisek — CHANGES_REQUESTED** [review]

File: src/TableColumn/ActionButtons.php (hunk 3 above)
Comment (georgehristov) [reply src/TableColumn/ActionButtons.php]: It works like it is. I did not dig deeper enough to see why it works :)
I also prefer array_merge

**Review by mvorisek — APPROVED** [review]
Perfect :)

**Pull request merged**